### DISCUSSION TITLE
Rationale of "Safety trainings/Onboarding Phases"

### DISCUSSION THREAD

**Discussion by r-xyz** [usnistgov/NEMO#220] [Q&A]
Hi all,
Thanks once more for such a great tool.

We were trying to implement some of the new NEMO features but wanted to ensure our usage is as much as possible aligned with the vision behind such features. This would potentially save us time, avoiding to manually migrate training records in the future to the proper system.

Up until recently we were tracking the "General Facility H&S Induction", required for granting physical access, as a separate tool training. We used a hidden tool (named as the facility itself) and registered trainings for that when the user completed the induction.
This also enabled us to use the ["Email Grant Access" feature ](https://gitlab.com/nemo-community/nemo-ce/-/wikis/Features/Email-grant-access) in [Nemo-CE](https://gitlab.com/nemo-community/nemo-ce).

Unfortunately, since some versions ago, it is now impossible to register a training (or to schedule it in NEMO-CE) for an hidden tool. This is an annoying but minor issue, which can still be worked around by showing the tool while preventing users to make reservations using strict policies.

We saw that two new features had been introduced: "Safety Trainings" and "Onboarding Phases". Any of those would probably be conceptually more appropriate to register "General Facility Induction" instead of having a "fake" tool which might confuse users. We have a few questions regarding their usage and future development ideas:

1. Currently both options are simply a tick option under the User page. There is no way to track when the "Safety Training/Onboarding phase" was completed. Only exception might be using auditlog, still with no possibility of adding pre-existing records with an associated date. Is there any plan to extend training feature (both basic training recordinging and/or [NEMO-CE scheduling](https://gitlab.com/nemo-community/nemo-ce/-/wikis/Features/New-training-module)) to any of Safety trainings/Onboarding, or is that completely out of those features scope?
2. Related to above, is there any plan to add "Grant Badge Reader Access" to any of those features?
3. Is there any other suggested usage of NEMO's features to achieve a similar functionality? (Including grant access and training events on NEMO-CE).
4. If none of the above, would it be possible to revert to the possibility to train (and schedule training events on NEMO CE) for hidden tools? This way we could keep it hidden while using a coherent system to track both H&S and tools trainings.

Thanks in advance for any feedback :)

PS: the link for `Email Grant Access` in [NEMO-CE features wiki](https://gitlab.com/nemo-community/nemo-ce/-/wikis/Features) needs to be updated from `-/wikis/Features/Email-grant-badge-reader-access` to `-/wikis/Features/Email-grant-access` since the page was renamed. I wanted to suggest a change but access is restricted.

**Comment by rptmat57** [comment]
Hi,
I'll give you my best answers to those questions:

1. No, there are no plans currently to extend the functionality beyond the simple checkboxes. Auditlog is currently the best way to track the dates for those.
2. No, not at the moment
3. Can't think of anything. The way you are using it is very clever, I will have to remember that. The only other somewhat similar functionality is the facility tutorial, but that's global (prevents anyone from doing anything in NEMO) and not just for physical access.
4. Should be doable, probably would want it as a customization setting since this a bit of a tricky use of the feature, but I really like it (if you want to submit a PR, it should be relatively straightforward).

Thanks for reporting the dead link, I updated it!

**Reply by r-xyz** [reply]
Hi Mathieu,
Thanks for your reply.

1&2.  Sounds ok, that was also my understanding but I wanted to double check before submitting a PR.
3. I agree, facility rules is good for part of our need, but having full training capability gives the possibility to have more control and (in nemo-ce) to also schedule training/request physical access.
4. I submitted #221; I would like to submit a similar PR for nemo-ce, but I am not sure if I should wait for NEMO changes to be merged first (part of them are common).

**Reply by rptmat57** [reply]
since the changes do not affect models/migrations, you can submit a PR in nemo-ce with the exact changes you have here (once approved), plus the other nemo-ce specific changes. I do this quite often.
